pandas/core/series.py

```diff
@@ -431,13 +431,21 @@ def nonzero(self):
 
     def put(self, *args, **kwargs):
         """
-        return a ndarray with the values put
+        Applies the `put` methods to its `values` attribute
+        and returns the resulting ndarray. Please refer to
+        `numpy.ndarray.put` for full documentation.
 
         See also
         --------
         numpy.ndarray.put
         """
-        self._values.put(*args, **kwargs)
+        try:
+            self._values.put(*args, **kwargs)
+        except AttributeError:
+            klass = type(self._values).__name__
```

Comment:
Looking at it again, I'm not sure if this `try`-`except` is the best move to make.  This probably should be separated from this PR, but I added this specifically because `Categorical` (when used as the `_value` attribute) has no `put` attribute.  So is it better to add a `put` method for `Categorical` that raises `NotImplemented` or do something more general like this?